tensorflow/python/ops/math_ops.py

```diff
@@ -1788,18 +1788,24 @@ def _AddNShape(op):
 def _SelectShape(op):
```

Comment:
yes, math_ops.cc.  Should be tested in math_ops_test.cc.  The shape function in C++ should be similar to the python one, it's just more verbose cause it's not python.